app/lib/methods/sendMessage.ts

```diff
@@ -72,41 +75,42 @@ export async function resendMessage(message, tmid) {
 			_id: message.id,
 			rid: message.subscription.id,
 			msg: message.msg
-		};
+		} as TMessageModel;
 		if (tmid) {
 			m = {
 				...m,
 				tmid
-			};
+			} as TMessageModel;
 		}
 		m = await Encryption.encryptMessage(m);
-		await sendMessageCall.call(this, m);
+
+		await sendMessageCall(m);
 	} catch (e) {
 		log(e);
 	}
 }
 
-export default async function (rid, msg, tmid, user, tshow) {
+export default async function (rid: string, msg: string, tmid: string, user: IUser, tshow?: boolean) {
```

Comment:
`IUser` is wrong here.
We're actually expecting id and token only.

Reply:
We use `username` and `name` as well 👀 
```ts
tm.u = {
    _id: user.id || '1',
    username: user.username,
    name: user.name
};
```